python/cog/director/director.py

```diff
@@ -165,7 +170,16 @@ def _loop(self):
 
             try:
                 log.info("received message")
-                self._handle_message(message_id, message_json)
+                span_attributes = {
+                    "idling_ms": (
+                        datetime.datetime.utcnow() - started_idling_at
+                    ).total_seconds()
```

Comment:
To get millisecond precision I think we'd need  `(datetime.datetime.utcnow() - started_idling_at) / timedelta(microseconds=1)`

Reply:
It's right, it's because `total_seconds(0)` returns a float with ms precision. tbh copilot gave me this, I tested it, and it was fine 👍🏻